### PR TITLE
fix: uses nextest in the Release CI

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

I noticed from the discussion on https://github.com/GreptimeTeam/greptimedb/discussions/1570 that this could be the cause of the test failure in the release CI. This PR
- Change Makefile to use nextest
- Install nextest in corresponding jobs

I'm afraid our unit test can only run by nextest for now, due to those global variables.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -33,13 +33,9 @@ docker-image: ## Build docker image.
...
 ##@ Test
 
-.PHONY: unit-test
-unit-test: ## Run unit test.
-	cargo test --workspace
-
-.PHONY: integration-test
-integration-test: ## Run integation test.
-	cargo test integration
+.PHONY: nextest
```
Comment (zyy17) [inline Makefile:36]: suggestion: I think we need to add the `nextest` target to install the toolchain and use the `tests` target:

```suggestion
tests: nextest ## Run unit and integration tests.
	cargo nextest run

.PHONY: nextest ## Install nextest tools.
nextest:
	cargo --list | grep nextest || cargo install cargo-nextest --locked
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -84,13 +84,14 @@ jobs:
...
+        uses: taiki-e/install-action@nextest
       - name: Output package versions
         run: protoc --version ; cargo version ; rustc --version ; gcc --version ; g++ --version
 
-      # - name: Run tests
-      #   if: env.DISABLE_RUN_TESTS == 'false'
-      #   run: make unit-test integration-test sqlness-test
+      - name: Run tests
+        if: env.DISABLE_RUN_TESTS == 'false'
+        run: make nextest sqlness-test
```
Comment (killme2008) [inline .github/workflows/release.yml:94]: do we need to run the `integration-test` too?

**Review by killme2008 — APPROVED** [review]
A minor suggestion. LTGM

**Comment by killme2008** [comment]
@zyy17 PTAL

**Review by zyy17 — APPROVED** [review]
LGTM

**Pull request merged**